google/cloud/bigtable/internal/rowreaderiterator.cc

```diff
@@ -41,14 +41,14 @@ RowReaderIterator& RowReaderIterator::operator++() {
 }
 
 void RowReaderIterator::Advance() {
-  auto status_or_optional_row = owner_->Advance();
-  if (!status_or_optional_row) {
-    row_ = StatusOr<bigtable::Row>(std::move(status_or_optional_row).status());
+  auto variant = owner_->Advance();
+  if (absl::holds_alternative<bigtable::Row>(variant)) {
```

Comment:
You could do this with a visitor (but double check my ref-ref):

```cc
  struct Visitor {
    StatusOr<bigtable::Row> operator()(Status&& s) { return s; }
    StatusOr<bigtable::Row> operator()(bigtable::Row&& r) { return r; }
  };
  row_ = absl::visit(Visitor{}, owner_->Advance());
}
```

Reply:
Won't Fix. I am going to delete the class shortly.

Also... what happens if your code is given an OK Status? 🤔